api/client/src/main/java/org/projectnessie/client/http/NessieHttpClientBuilder.java

```diff
@@ -56,6 +56,20 @@ public interface NessieHttpClientBuilder extends NessieClientBuilder {
   @CanIgnoreReturnValue
   NessieHttpClientBuilder withResponseFactory(HttpResponseFactory responseFactory);
 
+  /**
+   * Add a request filter to the client. Enables low-level access to the request/response
+   * processing.
+   */
+  @CanIgnoreReturnValue
+  NessieHttpClientBuilder addRequestFilter(RequestFilter filter);
```

Comment:
nit: `withRequestFilter` maybe?... just to follow existing naming patterns.

Reply:
But you can add more than one filter.

Reply:
hmmm.... good point :)